auth_server/src/auth.rs

```diff
@@ -0,0 +1,40 @@
+use actix_web::{
+    dev::ServiceRequest, error::ErrorUnauthorized, http::header::AUTHORIZATION, Error,
+};
+use jsonwebtoken::{decode, DecodingKey, Validation};
+
+pub struct AuthMiddleware;
+
+impl AuthMiddleware {
+    pub fn validate_token(
+        req: ServiceRequest,
+        secret: &str,
+    ) -> Result<ServiceRequest, (Error, ServiceRequest)> {
+        println!("middleware");
+        dbg!(req.headers().get(AUTHORIZATION));
+        let auth_header = if let Some(header) = req.headers().get(AUTHORIZATION) {
+            if let Ok(auth_str) = header.to_str() {
+                auth_str
+            } else {
+                return Err((ErrorUnauthorized("Invalid authorization header"), req));
+            }
+        } else {
+            return Err((ErrorUnauthorized("Missing authorization header"), req));
+        };
```

Comment:
I believe we can reduce nesting here:
```
let auth_header = req.headers().get(AUTHORIZATION).ok_or_else(|| Error("Invalid authorization header"));
let auth_str = auth_header.to_str().map_err(|err| Error("Invalid authorization header: {err}"))
```

Reply:
ok_or_else is returning Result<> type and I couldn't use to_str()
will investigate more on this